### ISSUE TITLE
Fichier de langues non lisible

### ISSUE THREAD

**Issue by MaxNicol** [guyfrancoeur/INF3135_E2019_TP#35]
Bonjour,

Je me questionne sur le fonctionnement voulu pour l'option  -l lorsqu'un fichier dans le répertoire a une extension valide (.fr, .en ou .de), mais n'est pas ouvrable ou lisible. Si tous les fichiers ne sont pas lisible, on retourne clairement un code d'erreur, mais qu'arrive-t-il si certains le sont et d'autres ne le sont pas?

Présentement, je retourne un code d'erreur dès qu'un fichier du répertoire passé en paramètre n'est pas ouvrable ou lisible, mais il pourrait être voulu de simplement utiliser ceux qui le sont et d'ignorer ceux qui ne le sont pas.

Je suis curieux de savoir comment vous gérez cette possibilité.

**Comment by emilepharand** [comment]
Bonjour,

Dans le cas où certains fichiers du répertoire modèle sont valides et que d'autres ne le sont pas, personnellement je lis ceux-là et j'ignore simplement ceux qui ne le sont pas.

**Comment by AxelSally2** [comment]
Bonjour,

Je pense tu devrais retourner un code d'erreur car le -b se fera en partie. Supposons que tu as le modèle `test.fr` dans ton répertoire de modèle et qui est le modèle qui te permet de déterminer la clé mais que celui-ci n'est pas ouvrable. Ton `-b` va utiliser les autres modèles si tu ignores `test.fr` mais il ne trouvera pas la bonne clé donc tu ne trouveras pas le bon message. Si tu retournes un code d'erreur, tu réduis les chances que ton `-b` se trompe.

**Comment by emilepharand** [comment]
Oui, tout à fait! Désolé, je croyais que la question concernait des fichiers qui auraient une extension autre que ".fr", ".en" ou ".de" dans le répertoire modèle. Si un fichier a une extension valide, mais n'est pas accessible à cause d'une erreur, il faut certainement retourner un code d'erreur. :)

**Comment by AxelSally2** [comment]
Content d'avoir pu aider. Bon Courage
